project/GenTupleAggregators.scala

```diff
@@ -66,8 +66,17 @@ object MapAggregator {
   }
 
   def genMapMethods(max: Int, isMonoid: Boolean = false): String = {
+    val aggType = if (isMonoid) "Monoid" else ""
+
+    // there's no Semigroup[Tuple1[T]], so just use T as intermediary type instead of Tuple1[T]
+    val aggregatorForOneItem = s"""
+       |def apply[A, B, C](aggDef: (String, ${aggType}Aggregator[A, B, C])): ${aggType}Aggregator[A, B, Map[String, C]] = {
+       |  val (key, agg) = aggDef
+       |  agg.andThenPresent(value => Map(key -> value))
+       |}
+    """.stripMargin
+
     (2 to max).map(i => {
```

Comment:
it didn't seem as simple as changing `2` to `1`...
with all these `%s` formatting, it's too hard infer anything about what the code-gen below does. so I'd leave it as is, at least until the original `2-max` part is refactored to use string interpolation (at least).